### PR TITLE
Add the Dapr /healthz API spec

### LINKED ISSUE #455: Add the new Dapr healthz api specification doc
As the new Dapr /healthz API is added, this issue is to add the API spec document.

Feature Reference : 
https://github.com/dapr/dapr/issues/1288

### PR DESCRIPTION
**Pull request by shalabhms**
# Description

As the new Dapr /healthz API is added, this issue is to add the API spec document.

Feature Reference :
dapr/dapr#1288

## Issue reference

Closes #455 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [X] Extended the documentation

### CODE REVIEW

File: reference/api/health_api.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Parameter | Description
+--------- | -----------
+daprPort | The Dapr port.
+
+#### Examples
+
+```shell
+curl http://localhost:3500/v1.0/healthz
+  -H "Content-Type: application/json"
+```
```
Comment (youngbupark) [inline reference/api/health_api.md]: 200 OK is supposed to have request body. I wonder if we need to return simple body. e.g.

```
{"status":"ok"}
```

@yaron2 @amanbha how do you think ?

Comment (yaron2) [reply reference/api/health_api.md]: We need to remove the header with the content type.  We don't have to return a response, a status code with no body is valid, this is what Kubernetes health checks expect also. They only examine the status code.

Comment (shalabhms) [reply reference/api/health_api.md]: Thanks, removed the header with the content type.

File: reference/api/health_api.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+```http
+GET http://localhost:<daprPort>/v1.0/healthz
+```
+
+#### HTTP Response Codes
+
+Code | Description
+---- | -----------
+200  | health is ok
```
Comment (yaron2) [inline reference/api/health_api.md]: I would change to `dapr is healthy` and `dapr is not healthy`

Comment (yaron2) [reply reference/api/health_api.md]: Other than that, LGTM

Comment (shalabhms) [reply reference/api/health_api.md]: Done as suggested. thanks.

**Pull request merged**